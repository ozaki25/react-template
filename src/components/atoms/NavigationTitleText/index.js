import React from 'react';
import PropTypes from 'prop-types';
import P from '../P';

function NavigationTitleText({ children }) {
  return <Text>{children}</Text>;
}

export const Text = P.extend`
  color: #333;
  font-size: 18px;
  font-weight: bold;
`;

NavigationTitleText.propTypes = {
  children: PropTypes.node.isRequired,
};

export default NavigationTitleText;
