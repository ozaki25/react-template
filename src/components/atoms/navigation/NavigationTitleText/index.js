import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function NavigationTitleText({ children }) {
  return <P>{children}</P>;
}

export const P = styled.p`
  color: #333;
  font-size: 18px;
  font-weight: bold;
`;

NavigationTitleText.propTypes = {
  children: PropTypes.node.isRequired,
};

export default NavigationTitleText;
