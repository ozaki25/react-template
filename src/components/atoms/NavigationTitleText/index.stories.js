import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';

import NavigationTitleText from '.';

const stories = storiesOf('atoms', module);

stories.add(
  'NavigationTitleText',
  withInfo('ナビゲーションバーのタイトルで使うテキスト')(() => (
    <NavigationTitleText>{text('title', 'タイトル')}</NavigationTitleText>
  )),
);
