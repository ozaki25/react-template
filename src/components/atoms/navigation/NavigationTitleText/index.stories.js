import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';

import NavigationTitleText from '.';

const stories = storiesOf('atoms/navigation', module);

stories.add(
  'NavigationTitleText',
  withInfo('ナビゲーションバーのタイトルで使うテキスト')(() => (
    <NavigationTitleText>{text('title', 'タイトル')}</NavigationTitleText>
  )),
);
