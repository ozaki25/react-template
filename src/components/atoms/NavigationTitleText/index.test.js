import React from 'react';
import { snapshot } from '../../../../test/helpers';
import NavigationTitleText from './';

snapshot('NavigationTitleText', <NavigationTitleText>タイトル</NavigationTitleText>);
