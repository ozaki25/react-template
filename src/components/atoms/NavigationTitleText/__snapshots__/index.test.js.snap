// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NavigationTitleText 1`] = `
<styled.p>
  タイトル
</styled.p>
`;
